Modules/Registration/Common/include/itkImageToImageMetric.hxx

```diff
@@ -1329,7 +1329,7 @@ ImageToImageMetric< TFixedImage, TMovingImage >
 ::GetValueAndDerivativeMultiThreadedPostProcessInitiate() const
 {
   m_Threader->SetSingleMethod( GetValueAndDerivativeMultiThreadedPostProcess,
-                               (void *)( &m_ThreaderParameter ) );
+                               const_cast< void * >( static_cast< const void * >( &m_ThreaderParameter ) ) );
```

Comment:
Note:  This is exactly what is done several other times in this same file a few lines above.  Just propagating the same solution here.